xonsh/tools.py

```diff
@@ -1032,7 +1034,28 @@ def print_exception(msg=None, exc_info=None):
                 "xonsh: To log full traceback to a file set: "
                 "$XONSH_TRACEBACK_LOGFILE = <filename>\n"
             )
-        traceback.print_exception(*exc_info, limit=limit, chain=chain)
+
+        traceback_str = "".join(
+            traceback.format_exception(*exc_info, limit=limit, chain=chain)
+        )
+
+        # color the traceback if available
+        _, interactive = _get_manual_env_var("XONSH_INTERACTIVE", 0)
+        _, color_results = _get_manual_env_var("COLOR_RESULTS", 0)
+        _, shell_type = _get_manual_env_var("SHELL_TYPE")
+        if (
+            interactive
+            and color_results
+            and shell_type == "prompt_toolkit"
```

Comment:
does shell_type need to be ptk?

Reply:
Thank you, I have now removed this restriction. The original problem was that I did not import enough, and it only worked "by chance" on prompt_toolkit since it loaded the required submodules.